workflow/Snakefile

```diff
@@ -12,6 +16,9 @@ nseeds = config["nseeds"]
 start_seed = 100
 seeds = range(start_seed, start_seed + nseeds)
 
+hyperparams = config["hyperparams"] if "hyperparams" in config else None
```

Comment:
@sklucas If the "hyperparams" key is not in the config dictionary, then the value of the hyperparams is set to `None` (in R this would be equivalent to `NULL`). Without this if-else statement and if a user doesn't include "hyperparams" in the config file, a key error would be thrown when trying to access `config["hyperparams"]`. The if-else statement makes it optional to define "hyperparams" in the config file. All other config options are mandatory.

Reply:
Thank you! I get the purpose now.